app/contacts/contact_ftp.py

```diff
@@ -0,0 +1,188 @@
+import json
+import os
+import re
+import asyncio
+import aioftp
+import sys
+
+from app.utility.base_world import BaseWorld
+
+
+class Contact(BaseWorld):
+    def __init__(self, services):
+        self.name = 'ftp'
+        self.description = 'Accept agent beacons through ftp'
+        self.contact_svc = services.get('contact_svc')
+        self.file_svc = services.get('file_svc')
+        self.logger = BaseWorld.create_logger('contact_ftp')
+        self.host = self.get_config('app.contact.ftp.host')
+        self.port = self.get_config('app.contact.ftp.port')
+        self.directory = self.get_config('app.contact.ftp.server.dir').lstrip('/')
+        self.home = os.path.join('/', self.directory)
+        self.user = self.get_config('app.contact.ftp.user')
+        self.pword = self.get_config('app.contact.ftp.pword')
+        self.server = None
+
+    async def start(self):
+        self.set_up_server()
+        if sys.version_info >= (3, 7):
+            t = asyncio.create_task(self.ftp_server_python_new())
+        else:
+            t = asyncio.create_task(self.ftp_server_python_old())
+        await t
+
+    def set_up_server(self):
+        # Define a new user with full r/w permissions
+        user = (
+            aioftp.User(
+                str(self.user),
+                str(self.pword),
+                home_path=self.home,
+                permissions=(
+                    aioftp.Permission('/', readable=False, writable=False),
+                    aioftp.Permission(self.home, readable=True, writable=True),
+                ),
+                maximum_connections=256,
+                read_speed_limit=1024 * 1024,
+                write_speed_limit=1024 * 1024,
+                read_speed_limit_per_connection=100 * 1024,
+                write_speed_limit_per_connection=100 * 1024
+            ),
+            aioftp.User(
+                home_path='/anon',
+                permissions=(
+                    aioftp.Permission('/', readable=False, writable=False),
+                    aioftp.Permission('/anon', readable=True),
+                ),
+                maximum_connections=5,
+                read_speed_limit=1024 * 1024,
+                write_speed_limit=1024 * 1024,
+                read_speed_limit_per_connection=100 * 1024,
+                write_speed_limit_per_connection=100 * 1024
+            ),
+        )
+        # Instantiate FTP server on local host and listen on port indicated in config
```

Comment:
these comments are good candidates for functions

Reply:
I can, but I think it would be incredibly inefficient. It would become 1 method that declares a variable and a method that passes that variable to another class.

Reply:
I think if anything, this would apply more to the user setup - so maybe something like `users = self.setup_ftp_users()`, where `setup_ftp_users` creates and returns that user tuple